lib/pages/chat/events/download_video_widget.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:fluffychat/pages/chat/events/download_video_state.dart';
+import 'package:fluffychat/pages/chat/events/event_video_player.dart';
+import 'package:fluffychat/pages/chat/events/message_content_style.dart';
+import 'package:fluffychat/presentation/mixins/handle_video_download_mixin.dart';
+import 'package:fluffychat/presentation/mixins/play_video_action_mixin.dart';
+import 'package:fluffychat/utils/localized_exception_extension.dart';
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/utils/twake_snackbar.dart';
+import 'package:fluffychat/widgets/mxc_image.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_ref_colors.dart';
+import 'package:matrix/matrix.dart';
+
+class DownloadVideoWidget extends StatefulWidget {
+  final Event event;
+
+  const DownloadVideoWidget({super.key, required this.event});
+
+  @override
+  State<StatefulWidget> createState() => _DownloadVideoWidgetState();
+}
+
+class _DownloadVideoWidgetState extends State<DownloadVideoWidget>
+    with HandleVideoDownloadMixin, PlayVideoActionMixin {
+  final _downloadStateNotifier = ValueNotifier(DownloadVideoState.initial);
+  String? path;
+  final downloadProgressNotifier = ValueNotifier(0.0);
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      _downloadAction();
+    });
+    super.initState();
+  }
+
+  void _downloadAction() async {
+    _downloadStateNotifier.value = DownloadVideoState.loading;
+    try {
+      path = await handleDownloadVideoEvent(
+        event: widget.event,
+        playVideoAction: (path) => playVideoAction(
+          context,
+          path,
+          event: widget.event,
+        ),
+        progressCallback: (count, total) {
+          downloadProgressNotifier.value = count / total;
+        },
+      );
+      _downloadStateNotifier.value = DownloadVideoState.done;
+    } on MatrixConnectionException catch (e) {
+      _downloadStateNotifier.value = DownloadVideoState.failed;
+      TwakeSnackBar.show(
+        context,
+        e.toLocalizedString(context),
```

Comment:
dont show the raw error to user.

Reply:
it's not, the error go through the `toLocalizedString` method
<img width="566" alt="Screenshot 2023-11-15 at 11 06 30" src="https://github.com/linagora/twake-on-matrix/assets/43041967/8d22554a-3634-4546-8f4a-8c1697c6b3db">